src/idex.ts

```diff
@@ -48,25 +47,24 @@ export class IdentityResolver {
     this.generateIdCookie = this.idexConfig.idCookieMode === 'generated'
     this.peopleVerifiedId = nonNullConfig.peopleVerifiedId
 
-    this.tuples = []
-
-    this.tuples.push(...asStringParam('duid', nonNullConfig.peopleVerifiedId))
-    this.tuples.push(...asStringParam('us_privacy', nonNullConfig.usPrivacyString))
-    this.tuples.push(...asParamOrEmpty('gdpr', nonNullConfig.gdprApplies, v => encodeURIComponent(v ? 1 : 0)))
-    this.tuples.push(...asStringParamWhen('n3pc', nonNullConfig.privacyMode ? '1' : '0', v => v === '1'))
-    this.tuples.push(...asStringParam('gdpr_consent', nonNullConfig.gdprConsent))
-    this.tuples.push(...asStringParam('did', nonNullConfig.distributorId))
-    this.tuples.push(...asStringParam('gpp_s', nonNullConfig.gppString))
-    this.tuples.push(...asStringParam('gpp_as', nonNullConfig.gppApplicableSections?.join(',')))
-    this.tuples.push(...asStringParam('cd', nonNullConfig.cookieDomain))
-    this.tuples.push(...encodeIdCookieParam(nonNullConfig.resolvedIdCookie))
+    this.query = new QueryBuilder()
+      .addOptionalParam('duid', nonNullConfig.peopleVerifiedId)
+      .addOptionalParam('us_privacy', nonNullConfig.usPrivacyString)
+      .addOptionalParam('gdpr', onNonNull(nonNullConfig.gdprApplies, v => v ? 1 : 0))
+      .addOptionalParam('n3pc', nonNullConfig.privacyMode ? 1 : undefined)
+      .addOptionalParam('gdpr_consent', nonNullConfig.gdprConsent)
+      .addOptionalParam('did', nonNullConfig.distributorId)
+      .addOptionalParam('gpp_s', nonNullConfig.gppString)
+      .addOptionalParam('gpp_as', nonNullConfig.gppApplicableSections?.join(','))
+      .addOptionalParam('cd', nonNullConfig.cookieDomain)
+      .addOptionalParam('ic', encodeIdCookie(nonNullConfig.resolvedIdCookie))
```

Comment:
@3link should we send empty ones (failed resolution) to idex as well or is live-pixel enough?